airflow/providers/google/marketing_platform/operators/campaign_manager.py

```diff
@@ -298,6 +299,12 @@ def __init__(
         self.gcp_conn_id = gcp_conn_id
         self.delegate_to = delegate_to
 
+    def prepare_template(self) -> None:
+        # If .json is passed then we have to read the file
+        if isinstance(self.report, str) and self.report.endswith('.json'):
```

Comment:
This would work for templates passed in with a *.json extension, but doesn't cover the case where report is passed as a JSON string template.  For example:

```
create_report = GoogleCampaignManagerInsertReportOperator(
        task_id="create_report",
        gcp_conn_id=self.config.get("gcp_connection_id"),
        profile_id=self.config.get("cm_profile_id"),
        report="""
            {
               'name': '{{ params.title }}',
               'accountId': '{{ params.account }}',
               'fileName': '{{ params.file_name }}',
               'type': 'STANDARD',
               'criteria': {
               ...""",
        params=report_params,
        dag=self.dag)
```

Safest approach may be to parse the json string within the execute method.

Reply:
According to documentation user should provide a dictionary so I would say that passing `str` don't have to be supported.

Reply:
You can create a task as below to create a task based on a JSON file from another source e.g. variable.
```
create_report = GoogleCampaignManagerInsertReportOperator(
        task_id="create_report",
        gcp_conn_id=self.config.get("gcp_connection_id"),
        profile_id=self.config.get("cm_profile_id"),
        report=json.loads("""
            {
               'name': '{{ params.title }}',
               'accountId': '{{ params.account }}',
               'fileName': '{{ params.file_name }}',
               'type': 'STANDARD',
               'criteria': {
               ..."""),
        params=report_params,
        dag=self.dag)
```

Reply:
True, although you can't take advantage of Airflow's [built-in macros](https://airflow.apache.org/docs/stable/macros-ref.html) like task_instance.  Also deviates from the behavior of other templated operators (e.g. [BashOperator](https://github.com/apache/airflow/blob/master/airflow/operators/bash.py)).

Reply:
You can because the templates are rendered recursively.
https://github.com/apache/airflow/blob/1c48ffbe25c3e304660b7e75a49e88bd114dde46/airflow/models/baseoperator.py#L872-L929
The problem only occurs when you use Jinja templates to generate new structures e.g. create new array elements.